### DISCUSSION TITLE
runtime: epollwait on fd 4 failed with 9 fatal error: runtime: netpoll failed

### DISCUSSION THREAD

**Discussion by lwwgo** [panjf2000/gnet#409] [Q&A]
潘少你好，遇到一个问题，报错处没有明确是gnet引入的，但是代码中有关网络poll的只有gnet有用到。网上查了很多资料，没有靠谱的，希望帮忙看下。
环境：linux mips 32位
go版本：go1.19
出现频率：偶发问题，1/2天一次，每次都是fd 4 failed with 9
目的：使用gnet进行udp网络传输
报错现场：
runtime: epollwait on fd 4 failed with 9
fatal error: runtime: netpoll failed
runtime stack:
runtime.throw({0x9f10a5, 0x17})
	/usr/go/src/runtime/panic.go:1047 +0x54 fp=0x18516d4 sp=0x18516c0 pc=0x5bbd8
runtime.netpoll(0x0)
	/usr/go/src/runtime/netpoll_epoll.go:130 +0x548 fp=0x1851f24 sp=0x18516d4 pc=0x568e0
runtime.findRunnable()
	/usr/go/src/runtime/proc.go:2652 +0x3d8 fp=0x1851fb0 sp=0x1851f24 pc=0x66510
runtime.schedule()
	/usr/go/src/runtime/proc.go:3219 +0xe8 fp=0x1851fcc sp=0x1851fb0 pc=0x6847c
runtime.park_m(0x8203cc0)
	/usr/go/src/runtime/proc.go:3370 +0x1b8 fp=0x1851fe0 sp=0x1851fcc pc=0x68c54
runtime.mcall()
	/usr/go/src/runtime/asm_mipsx.s:141 +0x50 fp=0x1851fe8 sp=0x1851fe0 pc=0x9d2d8

**Comment by panjf2000** [comment]
gnet 是哪个版本的？

这个堆栈错误表示的是 go 内部的 netpoll 调用 epoll_wait 系统调用失败

**Comment by panjf2000** [comment]
失败的源码位置：https://github.com/golang/go/blob/go1.19/src/runtime/netpoll_epoll.go#L129-L130

原因是 epoll_wait 系统调用被 OS 的 signal handler 打断了，具体原因未知，详情可以看：https://man7.org/linux/man-pages/man2/epoll_wait.2.html

**Reply by lwwgo** [reply]
gnet 1.6.6版本（从gnet库fork出来的：https://github.com/Ricardo-Evans/gnet/tree/v1.7.4）。
源码位置也找到了，不过一直没看出来原因，所以发贴求助，感谢回复。我再研究研究
